test/on_error_test.rb

```diff
@@ -0,0 +1,46 @@
+require 'test_helper'
+
+# The on_error call is tested herer.
+
+describe 'on_error' do
+  it "is called when there is a missing argument" do
+    opts = Slop::Options.new
+    opts.string "-n", "--name"
+    opts.on_error do |ex|
+      raise ArgumentError.new(ex.to_s)
```

Comment:
I'm not sure that raising is the best idea for this. If the feature stops working, the "rescue" won't have triggered and the `assert_match` will be ignored, but the tests will continue to pass. Maybe add an outer variable and then capture the exception and assert some behaviour of the exception object?

Reply:
Ok.